npm/npm_test.go

```diff
@@ -28,31 +30,29 @@ func getKey(obj interface{}, t *testing.T) string {
 	return key
 }
 
-func newNPMgr(t *testing.T) *NetworkPolicyManager {
+func newNPMgr(t *testing.T, exec utilexec.Interface) *NetworkPolicyManager {
 	npMgr := &NetworkPolicyManager{
+		Exec:             exec,
 		NsMap:            make(map[string]*Namespace),
 		PodMap:           make(map[string]*NpmPod),
 		TelemetryEnabled: false,
 	}
 
 	// This initialization important as without this NPM will panic
-	allNs, _ := newNs(util.KubeAllNamespacesFlag)
+	allNs, _ := newNs(util.KubeAllNamespacesFlag, npMgr.Exec)
 	npMgr.NsMap[util.KubeAllNamespacesFlag] = allNs
 	return npMgr
 }
 
 func TestMain(m *testing.M) {
 	metrics.InitializeAll()
 	iptMgr := iptm.NewIptablesManager()
-	iptMgr.Save(util.IptablesConfigFile)
+	iptMgr.UninitNpmChains()
```

Comment:
if a test fails, subsequent tests acn fail in the restore after test execution, in my opinion uniniting iptm and ipsm is better for the interim until everything is fakeexec'd out

Reply:
Can you elaborate this since I guess you will work on pluggable approach to run real-exec and fake-exec? Here, it sounds like that you use fakeexe in all UTs.

Reply:
eventually fakeexec will be used in all UT's but ns and netpol controller are still using a realexec